web/packages/shared/utils/highbar.ts

```diff
@@ -0,0 +1,273 @@
+      timeSinceLastInvoke = time - lastInvokeTime;
+
+    // Either this is the first call, activity has stopped and we're at the
+    // trailing edge, the system time has gone backwards and we're treating
+    // it as the trailing edge, or we've hit the `maxWait` limit.
+    return (
+      lastCallTime === undefined ||
+      timeSinceLastCall >= wait ||
+      timeSinceLastCall < 0 ||
+      (maxing && timeSinceLastInvoke >= maxWait)
+    );
+  }
+
+  function timerExpired() {
+    var time = Date.now();
+    if (shouldInvoke(time)) {
+      return trailingEdge(time);
+    }
+    // Restart the timer.
+    timerId = setTimeout(timerExpired, remainingWait(time));
+  }
+
+  function trailingEdge(time) {
+    timerId = undefined;
+
+    // Only invoke if we have `lastArgs` which means `func` has been
+    // debounced at least once.
+    if (trailing && lastArgs) {
+      return invokeFunc(time);
+    }
+    lastArgs = lastThis = undefined;
+    return result;
+  }
+
+  function cancel() {
+    if (timerId !== undefined) {
+      clearTimeout(timerId);
+    }
+    lastInvokeTime = 0;
+    lastArgs = lastCallTime = lastThis = timerId = undefined;
+  }
+
+  function flush() {
+    return timerId === undefined ? result : trailingEdge(Date.now());
+  }
+
+  function debounced() {
+    var time = Date.now(),
+      isInvoking = shouldInvoke(time);
+
+    lastArgs = arguments;
+    lastThis = this;
+    lastCallTime = time;
+
+    if (isInvoking) {
+      if (timerId === undefined) {
+        return leadingEdge(lastCallTime);
+      }
+      if (maxing) {
+        // Handle invocations in a tight loop.
+        timerId = setTimeout(timerExpired, wait);
+        return invokeFunc(lastCallTime);
+      }
+    }
+    if (timerId === undefined) {
+      timerId = setTimeout(timerExpired, wait);
+    }
+    return result;
+  }
+  debounced.cancel = cancel;
+  debounced.flush = flush;
+  return debounced;
+}
+
+interface MapCacheType {
+  delete(key: any): boolean;
+  get(key: any): any;
+  has(key: any): boolean;
+  set(key: any, value: any): this;
+  clear?: (() => void) | undefined;
+}
+
+type MemoizedFunction = {
+  cache: MapCacheType;
+};
+
+// Lift & Shift from lodash
+export function memoize<T extends (...args: any) => any>(
+  func: T
+): T & MemoizedFunction {
+  const memoized = function () {
+    const args = arguments;
+    const key = args[0];
+    const cache = memoized.cache;
+
+    if (cache.has(key)) {
+      return cache.get(key);
+    }
+    const result = func.apply(this, args);
+    memoized.cache = cache.set(key, result) || cache;
+    return result;
+  };
+  memoized.cache = new (memoize.Cache || MapCache)();
+  /* eslint-disable @typescript-eslint/ban-ts-comment*/
+  // @ts-ignore
+  return memoized;
+}
+
+// Expose `MapCache`.
+memoize.Cache = MapCache;
+
+function MapCache(entries?: any) {
+  let index = -1;
+  const length = entries == null ? 0 : entries.length;
+
+  this.clear();
```

Comment:
Connect failed to start because `clear` is not defined:
```
VM131 renderer_init:2 TypeError: this.clear is not a function
    at new MapCache (VM133 preload.js:38320:8)
    at memoize (VM133 preload.js:38308:20)
    at ./src/services/pty/ptyHost/resolveShellEnv.ts (VM133 preload.js:40341:92)
    at __webpack_require__ (VM133 preload.js:112250:42)
    at ./src/services/pty/ptyHost/buildPtyOptions.ts (VM133 preload.js:39889:74)
    at __webpack_require__ (VM133 preload.js:112250:42)
    at ./src/services/pty/ptyService.ts (VM133 preload.js:40448:82)
    at __webpack_require__ (VM133 preload.js:112250:42)
    at VM133 preload.js:112312:90
    at VM133 preload.js:112410:3
```

Reply:
Sorry about that, this has been resolved, Connect seems to work correctly now. Mind taking another look?

Reply:
I still have an issue, this time in `Hash`:
<img width="548" alt="image" src="https://user-images.githubusercontent.com/20583051/219670865-b964b6fd-a233-4649-9cee-c45a13f16d5b.png">